File: backend/src/main/java/com/uni/research/module/note/entity/Note.java
package com.uni.research.module.note.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * Note Entity
 */
@Data
@TableName("note")
public class Note implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * Note ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * User ID
     */
    private Long userId;

    /**
     * Document ID
     */
    private Long docId;

    /**
     * Note Content
     */
    private String content;

    /**
     * Note Type: DOCUMENT (document-level) or PARAGRAPH (paragraph annotation)
     */
    private String noteType;

    /**
     * Position Information (JSON format for paragraph notes)
     */
    private String position;

    /**
     * Tags (comma-separated)
     */
    private String tags;

    /**
     * Logic Delete
     */
    @TableLogic
    private Integer deleted;

    /**
     * Create Time
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * Update Time
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: frontend/src/api/note.ts
import request from './request'

// Note types
export interface NoteInfo {
    id?: number
    userId?: number
    docId: number
    content: string
    noteType?: string
    position?: string
    tags?: string
    createTime?: string
    updateTime?: string
}

// Note API services
export const noteAPI = {
    /**
     * Create note
     * Backend: POST /note
     */
    createNote(note: NoteInfo): Promise<any> {
        return request.post('/note', note)
    },

    /**
     * Update note
     * Backend: PUT /note/{id}
     */
    updateNote(id: number, note: NoteInfo): Promise<any> {
        return request.put(`/note/${id}`, note)
    },

    /**
     * Delete note
     * Backend: DELETE /note/{id}
     */
    deleteNote(id: number): Promise<any> {
        return request.delete(`/note/${id}`)
    },

    /**
     * Get document notes
     * Backend: GET /note/doc/{docId}
     */
    getDocumentNotes(docId: number): Promise<any> {
        return request.get(`/note/doc/${docId}`)
    },

    /**
     * Search notes
     * Backend: GET /note/search?keyword=xxx
     */
    searchNotes(keyword: string): Promise<any> {
        return request.get('/note/search', {
            params: { keyword }
        })
    }
}

export default noteAPI
